File: trunk/java/SR-RBD/src/ddb/restore/msg/RestoreTableList.java
package ddb.restore.msg;

import java.io.IOException;

import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;
import ddb.msg.MessageType;

public class RestoreTableList extends RestoreMessage {

	private TableVersion[] tables = null;
	
	public RestoreTableList()
	{
		// empty
	}
	
	public RestoreTableList(TableVersion[] tv)
	{
		setTables(tv);
	}
	
	public TableVersion[] getTables() {
		return tables;
	}
	
	public void setTables(TableVersion[] t) {
		tables = t;
	}
	
	
	@Override
	protected void fromBinary(DataInputStream s) throws IOException {
		int count = s.readInt();
		tables = new TableVersion[count];
		
		String name;
		int version;
		
		for(int i = 0; i < count; ++i)
		{
			name = s.readString();
			version = s.readInt();
			
			tables[i] = new TableVersion(name, version);
		}
	}
	
	@Override
	public MessageType getType() {
		return MessageType.RESTORE_TABLELIST;
	}

	@Override
	protected void toBinary(DataOutputStream s) throws IOException {
		// write length
		s.writeInt(tables.length);
		
		// write tables
		for(TableVersion tv : tables)
		{
			s.writeString(tv.getTableName());
			s.writeInt(tv.getVersion());
		}
		
	}

}

File: trunk/C#/RBD/RBD/Restore/Msg/RestoreTableList.cs
using System.Collections;
using System.Collections.Generic;
using RBD.DB;
using RBD.Communication;

namespace RBD.Restore.Msg
{
    public class RestoreTableList : RestoreMessage {

	    private List<TableVersion> tables = new LinkedList<TableVersion>();
    	
	    public RestoreTableList()
	    {
    		
	    }
    	
	    public RestoreTableList(List<TableVersion> list)
	    {
		    this.tables = list;
	    }
    	
	    public void setTables(List<TableVersion> tables) {
		    this.tables = tables;
	    }

	    public List<TableVersion> getTables() {
		    return tables;
	    }
    	
	    override public void FromBinary(DataInputStream s)  {
		    // read count
		    int count = s.ReadInt();
		    setTables(new LinkedList<TableVersion>());
    		
		    // read contents
		    for(int i = 0; i < count; ++i)
			    getTables().add(new TableVersion(s));
	    }

        override public MessageType GetMessageType()
        {
		    return MessageType.RESTORE_TABLELIST;
	    }

        override public void ToBinary(DataOutputStream s)
        {
		    // write length
		    s.writeInt(getTables().size());
    		
		    // write tables
		    foreach(TableVersion tv in getTables())
			    tv.toBinary(s);
	    }
    }
}
